llvm/lib/Target/DirectX/DXILRootSignature.cpp

```diff
@@ -75,31 +75,34 @@ static bool parseRootConstants(LLVMContext *Ctx, mcdxbc::RootSignatureDesc &RSD,
   if (RootConstantNode->getNumOperands() != 5)
     return reportError(Ctx, "Invalid format for RootConstants Element");
 
-  mcdxbc::RootParameter NewParameter;
-  NewParameter.Header.ParameterType =
+  dxbc::RootParameterHeader Header;
+  // this will be properly calculated when writing it.
+  Header.ParameterOffset = 0;
```

Comment:
I think this wording might be a little bit too brief. How about "The parameter offset doesn't matter here - we recalculate it during serialization"